pkg/cmd/codespace/common.go

```diff
@@ -150,23 +161,59 @@ func formatCodespacesForSelect(codespaces []*api.Codespace, includeOwner bool) [
 	return names
 }
 
+type getOrChooseCodespaceFilterOptions struct {
+	chooseCodespaceFilterOptions
+
+	CodespaceName string
+}
+
+var errInvalidGetOrChooseCodespaceCommandArgs = errors.New("invalid arguments: --codespace and --repo should not both be used together")
+
+// addGetOrChooseCodespaceCommandArgs adds command arguments for building getOrChooseCodespaceFilterOptions
+// to be passed into getOrChooseCodespace. It also adds pre-run validations on those arguments.
+func addGetOrChooseCodespaceCommandArgs(cmd *cobra.Command, opts *getOrChooseCodespaceFilterOptions) {
+	cmd.Flags().StringVarP(&opts.CodespaceName, "codespace", "c", "", "Name of the codespace")
+	cmd.Flags().StringVarP(&opts.Repo, "repo", "R", "", "Filter codespace selection by repository name (user/repo)")
+
+	existingPreRun := cmd.PersistentPreRunE
+	cmd.PersistentPreRunE = func(cmd *cobra.Command, args []string) error {
+		if err := validateGetOrChooseCodespaceCommandArgs(*opts); err != nil {
+			return err
+		}
+
+		if existingPreRun != nil {
+			return existingPreRun(cmd, args)
+		}
+
+		return nil
+	}
```

Comment:
Cobra has functionality to make flags mutually exclusive `cmd.MarkFlagsMutuallyExclusive("codespace", "repo")`. Or you can use `cmdutil.MutuallyExclusive` that we use in other commands.

Reply:
Oh cool! I swear I spent forever looking for that 😅